core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -880,15 +880,12 @@ public EnvVars getEnvironment(TaskListener log) throws IOException, InterruptedE
         if (ws!=null)   // if this is done very early on in the build, workspace may not be decided yet. see HUDSON-3997
             env.put("WORKSPACE", ws.getRemote());
 
-        project.getScm().buildEnvVars(this,env);
+        project.getScm().buildEnvironment(this, env);
 
         if (buildEnvironments!=null)
             for (Environment e : buildEnvironments)
                 e.buildEnvVars(env);
 
-        for (EnvironmentContributingAction a : getActions(EnvironmentContributingAction.class))
-            a.buildEnvVars(this,env,getBuiltOn());
```

Comment:
Even if `super` calls it, it may cause regressions in the existing logic. For example, if a whatever `EnvironmentContributingAction` relies on the `WORKSPACE` variable for nested resolution. Hence 🐜 

OTOH, not removing it also may lead to double resolution attempts, which may lead to interesting results to nested and especially recursive resolutions (e.g. `PATH=$PATH:foo`)

I have no good answer about how it should be handled. Environment resolution order in Jenkins is an undocumented mess flavored by a bunch of extension points. My proposal would be to not touch this invocation if possible